crates/biome_js_analyze/src/lint/nursery/no_useless_escape_in_string.rs

```diff
@@ -174,6 +174,10 @@ fn next_useless_escape(str: &str, quote: u8) -> Option<usize> {
                         }
                     }
                     _ => {
+                        // In template literals, \${ is a valid escape for producing a literal ${
+                        if quote == b'`' && c == b'$' && (matches!(it.next(), Some((_, b'{')))) {
```

Comment:
I think the second `it.next()` is problematic because you can miss cases like `\$\a`. Instead, you should clone the iterator before calling `next()`, i.e. `matches!(it..clone()next(), Some((_, b'{')))`.

Reply:
https://github.com/biomejs/biome/pull/6044 Thanks!